### PR TITLE
Ios build improvement

### PR DESCRIPTION
**Pull request by AntonioLangiu**
In this PR I fixed the ios build script. Now it's able to build all the dependencies, but it is not complete. We must fix the `build/dependency` script before we can complete the ios build script!

### CODE REVIEW

File: build/ios/scripts/build_arch.sh [hunk 1]
```diff
@@ -10,44 +10,69 @@ fi
...
-if [ "$PLATFORM" = "iPhoneOS" ]; then
-    EXTRA_CONFIG="--host=arm-apple-darwin14 --target=arm-apple-darwin14"
-fi
+OPT_FLAGS="-Os -g3"
+MAKE_JOBS=16
+
+ROOTDIR=$(cd $(dirname "$0") && pwd -P)
+echo $ROOTDIR
+SOURCEDIR=$(cd "${ROOTDIR}/../../../" && pwd -P)
+echo $SOURCEDIR
```
Comment (AntonioLangiu) [inline build/ios/scripts/build_arch.sh]: This are just for debug! They should be removed!

File: build/spec/all [hunk 2]
```diff
@@ -2,6 +2,6 @@
 set -e
 pkg_rootdir=$(cd $(dirname $(dirname $(dirname $0))) && pwd -P)
 cd $pkg_rootdir
-for dependency in jansson libevent libmaxminddb libressl; do
+for dependency in libressl jansson libevent libmaxminddb; do
```
Comment (bassosimone) [inline build/spec/all]: Can we keep libraries sorted?

**Comment by bassosimone** [comment]
I'd be very happy if this diff could be trimmed down so to avoid unnecessary changes into the diff, such that it's apparent what was actually changed!

File: build/ios/scripts/build.sh [hunk 3]
```diff
@@ -12,9 +12,9 @@ echo "Building for this architectures: $ARCHS"
 
 for ARCH in ${ARCHS}; do
     if [ "${ARCH}" == "i386" ] || [ "${ARCH}" == "x86_64" ]; then
-        PLATFORM="iPhoneSimulator"
+        PLATFORM="iphonesimulator"
```
Comment (AntonioLangiu) [inline build/ios/scripts/build.sh]: For @bassosimone: this is not just a variable renaming, we need lower case values to pass it to xcrun.
The name with CamelCase was used to retrieve the SDK directly with the folder, but I think using xcrun is more robust!

File: build/ios/scripts/build_arch.sh [hunk 4]
```diff
@@ -10,44 +10,66 @@ fi
...
+MIOS_VERSION="7.1"
+if [ $PLATFORM == "iphoneos" ]; then
+    MIOS_VERSION="-miphoneos-version-min=$MIOS_VERSION"
+else 
+    MIOS_VERSION="-mios-simulator-version-min=$MIOS_VERSION"
+fi
+
+HOST_FLAGS="-arch ${ARCH} ${MIOS_VERSION} -isysroot $(xcrun -sdk ${PLATFORM} --show-sdk-path)"
 
-DEVELOPER=$(xcode-select -print-path)
```
Comment (AntonioLangiu) [inline build/ios/scripts/build_arch.sh]: We don't need this because xcrun finds the correct directory

File: build/ios/scripts/build_arch.sh [hunk 5]
```diff
@@ -10,44 +10,66 @@ fi
...
 
-if [ "$PLATFORM" = "iPhoneOS" ]; then
-    EXTRA_CONFIG="--host=arm-apple-darwin14 --target=arm-apple-darwin14"
-fi
+OPT_FLAGS="-Os -g3"
+MAKE_JOBS=2
+
+ROOTDIR=$(cd $(dirname "$0") && pwd -P)
+SOURCEDIR=$(cd "${ROOTDIR}/../../../" && pwd -P)
+BUILDDIR=$(cd "${ROOTDIR}/../" && pwd -P)
```
Comment (bassosimone) [inline build/ios/scripts/build_arch.sh]: This change is purely cosmetic compared to the original. Can you revert it?

**Comment by bassosimone** [comment]
Hmm, you should start with rebasing this diff to master. There are things inside this diff that have already been committed to master.

**Comment by AntonioLangiu** [comment]
Closing this because I refactor all this changes in another branch! #523

**Pull request closed without merging**